sdk/identity/azure-identity/azure/identity/_internal/client_credential_base.py

```diff
@@ -13,6 +13,12 @@
 from .msal_credentials import MsalCredential
 
 
+def _get_known_kwargs(kwargs: Dict[str, Any]):
+    # Remove kwargs not expected by MSAL. These aren't typically passed by users, but this is a precaution.
+    known_kwargs = {"force_refresh", "authority", "correlation_id", "data"}
```

Comment:
rather than our layer of the stack validating this (silently eating these); shouldn't we let MSAL check this, fail, and throw an exception?

Reply:
Yea, the crux of the problem stems with how our functions and msal app functions accept arbitrary keyword args, but in the end, msal passes these keyword arguments to a function that _does not_ accept arbitary kwargs. So I do think it'd be worth opening an issue in MSAL regarding this since ideally MSAL should only pass in known kwargs. I can go ahead and open one.

For now, I think it's good to have this sanitation because we generally don't expect any additional kwargs to actually be passed in by a user. This is more just a preventative measure for _just in case_.